src/common/meta/src/ddl/drop_table/metadata.rs

```diff
@@ -29,6 +33,28 @@ impl DropTableProcedure {
         self.data.physical_region_routes = physical_table_route_value.region_routes;
         self.data.physical_table_id = Some(physical_table_id);
 
+        let table_info_value = self
```

Comment:
If there is a logical table (task.table_id != physical_table_id), can we avoid to do this check?